src/libopensc/cwa-dnie.c

```diff
@@ -711,8 +711,10 @@ static int dnie_transmit_apdu_internal(sc_card_t * card, sc_apdu_t * apdu)
 			if (tmp == SC_APDU_CASE_3_SHORT)
 				apdu->cse = SC_APDU_CASE_4_SHORT;
 			if (apdu->resplen == 0) {	/* no response buffer: create */
-				apdu->resp = buf;
-				apdu->resplen = 2048;
+				apdu->resp = calloc(1, 2*SC_MAX_APDU_BUFFER_SIZE);
```

Comment:
No I am not sure, and I should not have changed any functionality with this pull request. I should have stick to memory allocation things and avoid mistakes. I will change it back to 2048 if that number is right. I would prefer to use a constant with a meaningful name here, but obviously the correct constant. Thanks a lot for your comment!